File: 1297-maximum-number-of-balloons/1297-maximum-number-of-balloons.cpp
#include <unordered_map>
class Solution {
public:
    int maxNumberOfBalloons(string text) {
        unordered_map<char, int> dict;

        for (char val : text) {
            dict[val]++;
        }

        if (dict.find('b') == dict.end() || dict.find('a') == dict.end() || dict.find('l') == dict.end() || dict.find('o') == dict.end() || dict.find('o') == dict.end()) {
            return 0;
        }

        return min({dict['b'], dict['a'], dict['l']/2, dict['o']/2, dict['n']});
    }
};

File: 1297-maximum-number-of-balloons/1297-maximum-number-of-balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        words: dict[str, int] = {}
        for letter in text:
            words[letter] = words.get(letter, 0) + 1
        if not (words.get('b', 0) and words.get('a', 0) and words.get('l', 0) and words.get('o', 0) and words.get('n', 0)):
            return 0
        else:
            return min(words['b'], words['a'], words['l']//2, words['o']//2, words['n'])
